src/vcpkg/binarycaching.cpp

```diff
@@ -930,7 +930,7 @@ namespace
                     m_token_header,
                     m_accept_header.to_string(),
                     "Content-Type: application/octet-stream",
-                    "Content-Range: bytes 0-" + std::to_string(cache_size) + "/*",
+                    "Content-Range: bytes 0-" + std::to_string(cache_size - 1) + "/*",
```

Comment:
Needs to be guarded for `cache_size == 0`. Probably by another if around the if on line 927.

Also it seems like it would be a good idea to fill in the size based on https://developer.mozilla.org/docs/Web/HTTP/Headers/Content-Range ?

```suggestion
                    fmt::format("Content-Range: bytes 0-{}/{}", cache_size - 1, cache_size),
```